app/javascript/packs/shared/navigation/components/GlobalActivitiesModal.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { FormattedMessage } from "react-intl";
+import { Modal, Button } from "react-bootstrap";
+import _ from "lodash";
+
+import { getActivities } from "../../actions/ActivitiesActions";
+import ActivityElement from "./ActivityElement";
+import ActivityDateElement from "./ActivityDateElement";
+
+class GlobalActivitiesModal extends Component {
+  constructor(props) {
+    super(props);
+    this.displayActivities = this.displayActivities.bind(this);
+    this.addMoreActivities = this.addMoreActivities.bind(this);
+  }
+
+  displayActivities() {
+    if (this.props.activities.length === 0) {
+      return (
+        <li>
+          <FormattedMessage id="activities.no_data" />
+        </li>
+      );
+    }
+    return this.props.activities.map((activity, i, arr) => {
+      let newDate = new Date(activity.created_at);
```

Comment:
Maybe it would be better to use https://momentjs.com/ for dates. JavaScript has a lot of problems with dates